cfpb/cfgov-refresh
# Data migrations for site root and careers pages

This PR adds data migrations that create a site root, set up the careers page tree ("About Us", "Careers", etc.) and convert Django jobs into Wagtail `JobListingPage` pages.

Here's the rationale:

1. We want to use a Django data migration to convert existing `jobmanager.models.django.Job` model instances to Wagtail pages of type `jobmanager.models.pages.JobListingPage`.
1. To do this, we need a Wagtail parent page to attach these new child pages to.
1. It makes sense for this page to be the "Careers" root, and we need these pages to be created anyway, so let's add another Django data migration to make these.
1. All new pages need to be added to a site root, and it'd be good if this was consistent with how we set things up in production.
1. Let's make another Django data migration that properly sets up the Wagtail sites and site root in a way that is consistent with production.

If the code changes here seem a bit overly complex, it's because it's [not really possible](http://stackoverflow.com/a/33807804) to properly use Wagtail `Page` methods from within a Django migration. I had to cheat in one place, hopefully in a way that isn't problematic. But, most of the effort here was figuring out what is actually needed to programmatically add pages from within a migration.

Hopefully this PR allows:

1. Local databases and unit tests will have a site root that better matches what we have in production.
1. We can run this PR against the production code and the only side effect will be creation of (empty) careers pages (to be filled in by content editors manually) and migration of existing jobs (within the last 6 months) to the new Wagtail page format.
1. If this works, we can copy this model for future encoding of Wagtail content as data migrations.

## Additions

- New migration `v1.migrations.0008_site_root_data` that sets up site root as `CFGov` and also creates regular and content sites. This migration depends explicitly on the `DJANGO_STAGING_HOSTNAME` environment variable that we document in our `.env_SAMPLE` and appear to set in our various deploy JSON configs. It optionally allows for use of `DJANGO_HTTP_PORT` and defaults to port 80.
- New migration `jobmanager.migrations.0007_create_careers_pages` that sets up "About Us" (if it doesn't exist) along with the tree of Careers pages:
 - `/careers` (SublandingPage)
 - `/careers/working-at-cfpb` (BrowsePage)
 - `/careers/application-process` (BrowsePage)
 - `/careers/students-and-graduates` (BrowsePage)
 - `/careers/current-openings` (BrowsePage)
- New migration `jobmanager.migrations.0008_migrate_job_pages` that migrates existing Django `jobmanager.models.django.Job` objects to `jobmanager.models.pages.JobListingPage` pages.

This PR is a replacement for closed #2340.

## Removals

- Removed Events pages from `initial_data` script as they don't seem to be used by any tests and it's not clear why they are in there. Please let me know if there's a reason they need to exist there.

## Changes

- Added the new `v1.migrations.0008_site_root_data` migration to our Python unit test setup so that all unit tests can depend on having a `cfgov` root. Modified a few existing tests who relied on a site root named `home-page`.

## Testing

- Run unit tests both with database migrations and (`tox`) and without (`tox -e fast`).
- Testing with a local database:
 - Create a new database with `./drop-db.sh; ./create-mysql-db.sh; ./initial-data.sh`.
 - Log into the Wagtail admin and note that you have a `CFGov` root with both a `localhost` and `content.localhost` `Site` running on port 8000.
 - Also note that you have a set of (empty) careers pages.
- Testing with a production database:
 - Download a recent production data dump.
 - Browse around the Wagtail admin and note how the root is set up ("Live") and how there's an "About Us" page with no "Careers" page underneath it.
 - Run migrations against your local database (it should run these new `v1` and `jobmanager` migrations).
 - Note that a "Careers" page has been added with several pages underneath, including a whole bunch of recent job postings. All new pages should be in "Draft" state.
 - Try hitting [`/about-us/`](http://localhost:8000/about-us/careers/) and notice that pre-existing "About Us" content is not affected by this migration.
 - Try hitting [`/about-us/careers/`](http://localhost:8000/about-us/careers/careers) and [`/about-us/careers/current-openings`](http://localhost:8000/about-us/careers/current-openings) and note how these pages are still served from Django even though the pages exist in the Wagtail tree. This is desired behavior.

## Review

- @richaagarwal 

## Notes

- All migrations are wrapped in `@transaction.atomic` to prevent a database error from leaving the DB in an unpredictable state.

## Todos

- Add changes to CHANGELOG.

## Checklist
